conntracksync/watcher.go

```diff
@@ -86,7 +86,33 @@ func (ctw *ConntrackTableWatcher) doSync() error {
 			}
 		}
 		if c.PrimaryIp != "" && ctEntry.ReplySourceIP != c.PrimaryIp {
-			logrus.Infof("conntracksync: deleting mismatching conntrack entry found: %v. [expected: %v, got: %v]", ctEntry, c.PrimaryIp, ctEntry.ReplySourceIP)
+			logrus.Infof("conntracksync: deleting mismatching DNAT conntrack entry found: %v. [expected: %v, got: %v]", ctEntry, c.PrimaryIp, ctEntry.ReplySourceIP)
+			if err := conntrack.CTEntryDelete(ctEntry); err != nil {
+				logrus.Errorf("conntracksync: error deleting the conntrack entry: %v", err)
+			}
+		}
+	}
+
+	sCTEntries, err := conntrack.ListSNAT()
+	if err != nil {
+		logrus.Errorf("conntracksync: error fetching SNAT conntrack entries")
+		return err
+	}
+
+	for _, ctEntry := range sCTEntries {
+		var c *metadata.Container
+		var specificEntryFound, genericEntryFound bool
+		specificKey := ctEntry.ReplyDestinationIP + ":" + ctEntry.OriginalSourcePort + "/" + ctEntry.Protocol
```

Comment:
you are right, fixing it. I was looking at 4500:4500, so got confused. If it's something like 10001:80, it would have been obvious.